sphinxcontrib/confluencebuilder/translator/wiki.py

```diff
@@ -165,7 +180,10 @@ def depart_rubric(self, node):
         self.end_state()
 
     def visit_compound(self, node):
-        pass
+        if self.apply_hierarchy_children_macro:
+            self.add_text('{children:depth=%s}' % self.tocdepth)
+            self.add_text(self.nl)
+            raise nodes.SkipNode
```

Comment:
What does the skip node statement do?

Reply:
The purpose of the `nodes.SkipNode` is to drop any sub-processing of the compound type (i.e. the toc-tree wrapper). So, if `apply_hierarchy_children_macro` is configure, the child macro will be rendered in the document and no longer attempt to process the internals of the toctree. For the default case, the toctree type is entered and renders its internal elements using other `visit_*` calls (based on the internals of Sphinx). Since the method has a definition, there's no need to define a `pass`.